supervisor.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+
+set -e
+
+trap 'kill %1; kill %2' ERR EXIT
+
+cmd=$PWD/transfer2go
```

Comment:
This idea won't work here. Because you are passing command as an argument and then we are doing grep search on commands to check if the process is running or not. Ultimately it will return PID of this bash script only.